github/ci.go

```diff
@@ -0,0 +1,65 @@
+package github
+
+import (
+	"fmt"
+	"net/http"
+	"io/ioutil"
+	"encoding/json"
+)
+
+
+func  getBranchCIStatus(projectName string, owner string, branch string) (string, error) {
+	return getCIStatus(projectName, owner, branch, "")
+}
+
+func  getTagCIStatus(projectName string, owner string, tag string) (string, error) {
+	return getCIStatus(projectName, owner, "", tag)
+}
+
+func  getCIStatus(projectName string, owner string, branch string, tag string) (string, error) {
+
+
+	var url string
+	if branch == "" {
+	url = fmt.Sprintf("%s/repos/%s/%s/commits/%s/status", githubV3Url, owner, projectName, tag)
```

Comment:
This does not [url-encode](https://en.wikipedia.org/wiki/Percent-encoding), it will fail with edge cases.

Reply:
I explored a could of open-source projects and they do it the same way :) hence I picked this approach

Reply:
Yes, many projects do in a way that breaks. I encounter these bugs every now and then. It shouldn't be hard to find a url-encoding lib for Go, maybe it is even in the standard lib.

Reply:
I cannot quite get this functionality to work for now. I think we can go for this in further iterations.